msu/systems/mod_settings/settings_panel.nut

```diff
@@ -166,6 +166,34 @@
 		this.callSettingsFunction("flagDeserialize", [_in]);
 	}
 
+	function saveToPersistentData()
+	{
+		local data = {};
+		foreach (page in this.Pages)
+		{
+			foreach (setting in page.getSettings())
+			{
+				if (setting instanceof ::MSU.Class.AbstractSetting)
+				{
+					data[setting.getID()] <- setting.getValue();
+				}
+			}
+		}
+		this.getMod().PersistentData.createFile("ModSetting", data);
```

Comment:
This is very inefficient. I think we should have all settings get saved to a single file. otherwise, otherwise a user's save files will get cluttered up with files that do not need to be there when we can just map this to a `modID : Settings` table and then save/load that.